pymeasure/instruments/keysight/keysightDSOX1102G.py

```diff
@@ -0,0 +1,551 @@
+            - "CHAN": channel number (int)
+            - "OFFS": vertical offset (float)
+            - "RANG": vertical range (float)
+            - "COUP": "dc" or "ac" coupling (str)
+            - "IMP": input impedance (str)
+            - "DISP": currently displayed (bool)
+            - "BWL": bandwidth limiting enabled (bool)
+            - "INV": inverted (bool)
+            - "UNIT": unit (str)
+            - "PROB": probe attenuation (float)
+            - "PROB:SKEW": skew factor (float)
+            - "STYP": probe signal type (str)
+        """
+
+        # Using the instrument's ask method because Channel.ask() adds the prefix ":channelX:", and to query the
+        # configuration details, we actually need to ask ":channelX?", without a second ":"
+        ch_setup_raw = self.instrument.ask(":channel%d?" % self.number).strip("\n")
+
+        # ch_setup_raw hat the following format:
+        # :CHAN1:RANG +40.0E+00;OFFS +0.00000E+00;COUP DC;IMP ONEM;DISP 1;BWL 0;
+        # INV 0;LAB "1";UNIT VOLT;PROB +10E+00;PROB:SKEW +0.00E+00;STYP SING
+
+        # Cut out the ":CHANx:" at beginning and split string
+        ch_setup_splitted = ch_setup_raw[7:].split(";")
+
+        # Create dict of setup parameters
+        ch_setup_dict = dict(map(lambda v: v.split(" "), ch_setup_splitted))
+
+        # Add "CHAN" key
+        ch_setup_dict["CHAN"] = ch_setup_raw[5]
+
+        # Convert values to specific type
+        to_str = ["COUP", "IMP", "UNIT", "STYP"]
+        to_bool = ["DISP", "BWL", "INV"]
+        to_float = ["OFFS", "PROB", "PROB:SKEW", "RANG"]
+        to_int = ["CHAN"]
+        for key in ch_setup_dict:
+            if key in to_str:
+                ch_setup_dict[key] = str(ch_setup_dict[key])
+            elif key in to_bool:
+                ch_setup_dict[key] = (ch_setup_dict[key] == "1")
+            elif key in to_float:
+                ch_setup_dict[key] = float(ch_setup_dict[key])
+            elif key in to_int:
+                ch_setup_dict[key] = int(ch_setup_dict[key])
+        return ch_setup_dict
+
+
+class KeysightDSOX1102G(Instrument):
+    """ Represents the Keysight DSOX1102G Oscilloscope interface for interacting
+    with the instrument.
+
+    Refer to the Keysight DSOX1102G Oscilloscope Programmer's Guide for further details about
+    using the lower-level methods to interact directly with the scope.
+
+    .. code-block:: python
+        scope = KeysightDSOX1102G(resource)
+        scope.autoscale()
+        ch1_data_array, ch1_preamble = scope.download_data(source="channel1", points=2000)
+        # ...
+        scope.shutdown()
+
+     Known issues:
+        - The digitize command will be completed before the operation is. May lead to
+            VI_ERROR_TMO (timeout) occuring when sending commands immediately after digitize.
+            Current fix: if deemed necessary, add delay between digitize and follow-up command
+            to scope.
```

Comment:
This should fix the doc generation warning reported above.

```suggestion
    .. code-block:: python
    
        scope = KeysightDSOX1102G(resource)
        scope.autoscale()
        ch1_data_array, ch1_preamble = scope.download_data(source="channel1", points=2000)
        # ...
        scope.shutdown()

    Known issues:
    
    - The digitize command will be completed before the operation is. May lead to
      VI_ERROR_TMO (timeout) occuring when sending commands immediately after digitize.
      Current fix: if deemed necessary, add delay between digitize and follow-up command
      to scope.
```